gnc/sub8_system_id/sub8_system_id/estimate_drag_coefficients.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env python
+import rospy
+import yaml
+from geometry_msgs.msg import WrenchStamped
+from nav_msgs.msg import Odometry
+# Initialized ros parameters and global values (feel free to tweak, values were empirically gained through trial/error)
+
+
+class Drag_C_Calc(object):
+
+    def __init__(self):
+        self.linear_drag_x = 0
+        self.linear_drag_y = 0
+        self.linear_drag_z = 0
+        self.roll_drag = 0
+        self.pitch_drag = 0
+        self.yaw_drag = 0
+        self.velocity = 0
+        '''
+        Velocities between two time intervals will never be truly 'equal' in real world
+        Must compare the abs of the difference of the two with a small delta value
+        if the difference is less than this value we can conclude that the numbers are satisfyingly close
+        also used as a value that is satisfyingly close to zero. Hard-coded (can be changed).
+        '''
+        self.delta = .00001
+        # Max velocity cannot be initialized to 0 or there will be an initial divide-by-zero error
+        self.max_velocity = .1
+        self.bouyancy = 0
+        # Magnitude of applied force for determining drag coeffcients in linear axes
+        self.applied_linear_force = rospy.get_param("linear_force", default=10)
+        # Magnitude of applied torque for determining drag coeffcients in rotational axes
+        self.applied_torque = rospy.get_param("torque", default=5)
+        self.applied_force_down = rospy.get_param('force_down', default=-20)
+        self.time_of_apllied_force_down = rospy.get_param('time_of_force_down', default=10)
+        self.drag = {}
+
+    def find_bouyancy(self, choice):  # Initial function used upward force of buoyancy(used determining drag in Z axis)
+        down_force = -.5  # Applies initial downward force in z axis
+        pub = rospy.Publisher('/wrench', WrenchStamped, queue_size=30)
+        rospy.init_node('move_sub', anonymous=False)
+        force_msg = WrenchStamped()
+        force_msg.wrench.force.z = self.applied_force_down
+        pub.publish(force_msg)  # publish wrench with force
+        rospy.loginfo('Moving sub down...')
+        rospy.sleep(self.time_of_apllied_force_down)  # node sleeps for some amount of time before continuing
+        force_msg.wrench.force.z = 0  # Applies 0 force which stops downward force
+        pub.publish(force_msg)
+        while not (rospy.is_shutdown()):
+            rospy.Subscriber("/odom", Odometry, self.get_velocity, choice)
```

Comment:
Since you have these in multiple parts of your program, perhaps they should be a class variable